### PR TITLE
Enable storage of theme preferences and add submenu selection of theme

### PR DESCRIPTION
**Pull request by pyuxiang**
### Feature changes

Extension of theme toggling implementation proposed in #176.
Implements the requested enhancement to load user preferred theme, as well as a submenu option to toggle theme.
Default theme is also set to pastel.

image

### Additional changes (on top of #176)

- Changed the hardcoding of theme string values to an enumeration. This reduces JavaFX MainWindow dependency on the unusual `List<String>` construct.
- `GuiSettings` is passed the `Theme` enumeration during each theme toggle call. The Jackson library automatically handles serialization of the enumeration.

#### Data classes

- `GuiSettings` modified to accept additional theme parameter.
  Default is set to `Theme.PASTEL`. This remains a read-only construct.
- `Theme` is an enumeration, with static method to retrieve specified stylesheets for each theme.

#### Logic flow

1. `ToggleThemeCommand` called upon user input `theme`. Retrieves current theme from read-only `GuiSettings` in model, and returns a `CommandResult` with theme information.
2. `CommandResult` is overloaded for message passing of `Theme` value, similar to `exit` functionality. This is required so that the `MainWindow` itself is passed the change theme command to directly modify the UI.
3. `MainWindow` sets UI theme on load (in constructor) or upon passed `ToggleThemeCommand` (in `MainWindow::executeCommand`). Main logic for theme setting is in `MainWindow::setTheme`, which replaces the stylesheets. Added submenu bar in FXML for click-based theme modification.
4. Whenever theme is modified or when application is exit, the model's `GuiSettings` is replaced with a new (read-only) `GuiSettings` with updated theme, for `ToggleThemeCommand` to access and application to save theme preferences.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -258,6 +258,12 @@ Examples:
 * `help` lists all available commands.
 * `help list` shows the syntax and description for the `list` command.
 
+### Toggle theme : `theme`
+
+Toggles between Dark and Pink theme
```
Comment (garyljj) [inline docs/UserGuide.md:263]:
```suggestion
Toggles between Dark and Pastel theme
```

File: src/main/java/seedu/partyplanet/commons/core/GuiSettings.java [hunk 2]
```diff
@@ -16,23 +18,33 @@
...
 
     /**
      * Constructs a {@code GuiSettings} with the default height, width and position.
      */
     public GuiSettings() {
+        this(Theme.PASTEL);
+    }
+
+    /**
+     * Constructs a {@code GuiSettings} with the default height, width and position, with specified Theme.
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/commons/core/GuiSettings.java:31]:
```suggestion
     * Constructs a {@code GuiSettings} with the default height, width, position, and theme.
```

File: src/main/java/seedu/partyplanet/commons/core/GuiSettings.java [hunk 3]
```diff
@@ -16,23 +18,33 @@
...
+     */
+    public GuiSettings(Theme theme) {
         windowWidth = DEFAULT_WIDTH;
         windowHeight = DEFAULT_HEIGHT;
         windowCoordinates = null; // null represent no coordinates
+        this.theme = theme;
     }
 
     /**
      * Constructs a {@code GuiSettings} with the specified height, width and position.
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/commons/core/GuiSettings.java:41]:
```suggestion
     * Constructs a {@code GuiSettings} with the specified height, width, position, and theme
```

File: src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import seedu.partyplanet.ui.Theme;
+
+/**
+ * Toggles the theme of PartyPlanet between Dark and Pink.
+ */
+public class ToggleThemeCommand extends Command {
+
+    public static final String COMMAND_WORD = "theme";
+    public static final String MESSAGE_SUCCESS = "Changed to %s theme!";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Toggles PartyPlanet's theme between Dark and Pink.";
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java:13]:
```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Toggles PartyPlanet's theme between Dark and Pastel.";
```

File: src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        CommandResult commandResult;
+        Theme theme = model.getGuiSettings().getTheme();
+
+        switch (theme) {
+        case PASTEL:
+            commandResult = new CommandResult(String.format(MESSAGE_SUCCESS, "Dark"), false, Theme.DARK);
+            break;
+        case DARK: // fallthrough
+        default:
+            commandResult = new CommandResult(String.format(MESSAGE_SUCCESS, "Pink"), false, Theme.PASTEL);
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java:27]:
```suggestion
            commandResult = new CommandResult(String.format(MESSAGE_SUCCESS, "Pastel"), false, Theme.PASTEL);
```

File: src/main/java/seedu/partyplanet/ui/MainWindow.java [hunk 6]
```diff
@@ -196,11 +209,33 @@ private CommandResult executeCommand(String commandText) throws CommandException
...
+     * Set themes dynamically based on user command.
+     */
+    private void setTheme(Theme theme) {
+        scene.getStylesheets().clear();
+        scene.getStylesheets().addAll(Theme.getStyleSheets(theme));
+        logic.setGuiSettings(new GuiSettings(theme));
+        this.theme = theme;
+    }
+    @FXML
+    public void setThemePastel() {
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/ui/MainWindow.java:234]: Optional: can be private?
```suggestion
    private void setThemePastel() {
```

File: src/main/java/seedu/partyplanet/ui/MainWindow.java [hunk 7]
```diff
@@ -196,11 +209,33 @@ private CommandResult executeCommand(String commandText) throws CommandException
...
+        setTheme(Theme.PASTEL);
+    }
+    @FXML
+    public void setThemeDark() {
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/ui/MainWindow.java:238]: Optional: can be private?
```suggestion
    private void setThemeDark() {
```

**Review by garyljj — APPROVED** [review]
Standardized theme name. Change 'pink' to 'pastel'
Minor javadocs and visibility edits
Other than those, LGTM

File: src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package seedu.partyplanet.logic.commands;
+
+import seedu.partyplanet.model.Model;
+import seedu.partyplanet.ui.Theme;
+
+/**
+ * Toggles the theme of PartyPlanet between Dark and Pink.
```
Comment (garyljj) [inline src/main/java/seedu/partyplanet/logic/commands/ToggleThemeCommand.java:7]: Missed out in initial suggestions
```suggestion
 * Toggles the theme of PartyPlanet between Dark and Pastel.
```

**Comment by pyuxiang** [comment]
Great suggestions! All proposed changes have been integrated.

**Review by Ellevy — APPROVED** [review]
LGTM

**Pull request merged**